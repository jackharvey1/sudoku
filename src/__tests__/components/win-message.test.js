import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import WinMessage from '../../components/win-message';

let renderer;

beforeEach(() => {
    renderer = new ShallowRenderer();
});

describe('Win message component', () => {
    it('renders the win message as expected when sudoku is solved', () => {
        renderer.render(<WinMessage
            solved={true}
            resetFunction={() => {}}
        />);
        const output = renderer.getRenderOutput();
        expect(output).toMatchSnapshot();
    });

    it('returns null when the sudoku is not solved', () => {
        renderer.render(<WinMessage
            solved={false}
            resetFunction={() => {}}
        />);
        const output = renderer.getRenderOutput();
        expect(output).toMatchSnapshot();
    });
});
