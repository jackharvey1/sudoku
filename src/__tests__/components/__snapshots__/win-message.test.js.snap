// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Win message component renders the win message as expected when sudoku is solved 1`] = `
<div
  className="css-1w4wrbu"
>
  <div
    className="css-zkjlb6"
  >
    <p
      className="css-1yvtxxr"
    >
      Solved!
    </p>
    <button
      className="css-1u92zc4"
      onClick={[Function]}
    >
      Again?
    </button>
  </div>
</div>
`;

exports[`Win message component returns null when the sudoku is not solved 1`] = `null`;
